src/daemon.c

```diff
@@ -173,9 +176,23 @@ void delay_startup(void)
 	if(config.delay_startup == 0u)
 		return;
 
+	// Get uptime of system
+	struct sysinfo info = { 0 };
+	if(sysinfo(&info) == 0)
+	{
+		if(info.uptime > DELAY_UPTIME)
+		{
+			logg("Not sleeping as system has finished booting");
+			return;
+		}
+	}
+	else
+	{
+		logg("Unable to obtain sysinfo: %s (%i)", strerror(errno), errno);
+	}
+
 	// Sleep if requested by DELAY_STARTUP
-	logg("Sleeping for %d seconds as requested by configuration ...",
-	     config.delay_startup);
+	logg("Sleeping for %d seconds as requested by configuration ...", config.delay_startup);
 	sleep(config.delay_startup);
 	logg("Done sleeping, continuing startup of resolver...\n");
```

Comment:
```
[2022-05-19 20:19:42.293 1535M] Sleeping for 5 seconds as requested by configuration ...
[2022-05-19 20:19:47.293 1535M] Done sleeping, continuing startup of resolver...

[2022-05-19 20:19:47.438 1535M] Database version is 12
```


```suggestion
	logg("Done sleeping, continuing startup of resolver...");
```

Reply:
I believe this was intentional to get some more attention but this isn't really needed, probably.